### PR TITLE
Lift captured `project.toString` to conf. time.

### PR DESCRIPTION
**Pull request by Jeffset**
This change is required to fix gradle doc sample project configuration
cache compatibility.

Fixes: #21802

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by bamboo** [comment]
@Jeffset, please add your full name and github handle to the list of contributors in the release notes as described [here](https://github.com/gradle/gradle/blob/master/subprojects/docs/src/docs/release/notes.md?plain=1#L6).

File: subprojects/docs/src/snippets/buildlifecycle/projectEvaluateEvents/groovy/build.gradle [hunk 1]
```diff
@@ -2,10 +2,11 @@
 allprojects {
     afterEvaluate { project ->
         if (project.hasTests) {
-            println "Adding test task to $project"
+            def projectString = project.toString()
```
Comment (bamboo) [inline subprojects/docs/src/snippets/buildlifecycle/projectEvaluateEvents/groovy/build.gradle:5]: 👍

File: subprojects/docs/src/snippets/buildlifecycle/projectEvaluateEvents/kotlin/build.gradle.kts [hunk 2]
```diff
@@ -5,10 +5,11 @@ allprojects {
 
     afterEvaluate {
         if (extra["hasTests"] as Boolean) {
-            println("Adding test task to $project")
+            val projectString = project.toString()
```
Comment (bamboo) [inline subprojects/docs/src/snippets/buildlifecycle/projectEvaluateEvents/kotlin/build.gradle.kts:8]: 👍

**Comment by bamboo** [comment]
@bot-gradle test this

**Review by bamboo — APPROVED** [review]
🎉

**Comment by bamboo** [comment]
@bot-gradle test this

**Comment by Jeffset** [comment]
Judging by the logs of two failed builds, seems like restarting them would help. Probably a flaky timeout issue?

**Comment by bamboo** [comment]
@bot-gradle test and merge

**Pull request merged**